### PR TITLE
FR7 and FR 10

### LINKED ISSUE #7: FR7. View previous transactions
|   FR7. View previous transactions |
|-----------------------|
| Goal: FR7 Allows the user to look at their spending habits and learn how much they spend on certain things. |
| Stakeholders: All users |
| Description: When selecting an individual budget, the user will be taken to a new page, showing the budget at the top. The previous transactions are then listed below the budget. The user will be able to scroll down the page to view further previous transactions |
| Origin: Team meeting on 1/31 |
| Version: 1.0 | 
| Date: 01/31/2023 |
| Priority: 2 |

### LINKED ISSUE #10: FR10. Expanded Breakdown List
|  FR10. Expanded Breakdown List  |
|-----------------------|
|Goal: FR10 breaks down specific budget categories  |
| Stakeholders: All Users |
| Description: Selecting one of the breakdowns on the homepage will bring the user to another page that expands into a detailed list with every budget-specific purchase. |
| Origin: Team meeting on 1/31 |
| Version: 1.0 | 
| Date: 01/31/2023 |
| Priority: 2 |

### PR DESCRIPTION
**Pull request by amuwhao46**

### CODE REVIEW

File: lib/ExpensesPage/expensesContainer.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+import 'package:percent_indicator/linear_percent_indicator.dart';
+import 'package:budgie/ExpensesPage/expensesPageClass.dart';
```
Comment (Ahmed-Alagbari) [inline lib/ExpensesPage/expensesContainer.dart:3]: I'm pretty sure you don't have to import using the full path if they are in the same folder

File: lib/BudgetCategories/horProgressBar.dart [hunk 2]
```diff
@@ -50,7 +38,7 @@ class _HorProgressBar extends State<HorProgressBar> {
                 animateFromLastPercent: true,
                 lineHeight: 20,
                 percent: percentHelper(progressAmt, totalProgress),
-                progressColor: const Color.fromRGBO(34, 197, 94, 1),
+                progressColor: dynamicColor(progressAmt, totalProgress),
```
Comment (Ahmed-Alagbari) [inline lib/BudgetCategories/horProgressBar.dart:41]: Great job making the color dynamic for the progress bar

File: lib/ExpensesPage/expensesPage.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:percent_indicator/linear_percent_indicator.dart';
+import 'package:budgie/BudgetCategories/budgetCategoryPrice.dart';
+import 'package:budgie/ExpensesPage/expensesContainer.dart';
```
Comment (Ahmed-Alagbari) [inline lib/ExpensesPage/expensesPage.dart:5]: this also uses the full path, I don't think it's a big deal but maybe worth mentioning

**Review by Ahmed-Alagbari — COMMENTED** [review]
Looks pretty good to me! Approved

**Review by mxsingh — APPROVED** [review]
Looks good! Agree with Ahmed--I don't think it's necessary to 'import' the entire file path if it's in the same folder.

**Review by JayanthAnandNama — APPROVED** [review]
Looking good.

File: lib/BudgetCategories/budgetCategoryPrice.dart [hunk 4]
```diff
@@ -1,5 +1,30 @@
-class BudgetCategoryPrice {
-  String? budgetName;
-  int? currentCategoryPrice;
-  int? totalCategoryPrice;
+import 'package:flutter/material.dart';
+
+
+String category = 'Food';
+String progressAmt = '249';
```
Comment (Rey-Jey) [inline lib/BudgetCategories/budgetCategoryPrice.dart:5]: Are we going to start using test variables before or after sprint 2? wondering if we're supposed to begin making user inputs for this stuff

File: lib/BudgetCategories/budgetCategoryPrice.dart [hunk 5]
```diff
@@ -1,5 +1,30 @@
...
+// Calculates LinearPercentIndicator.percent
+double percentHelper(String dividend, String divisor) {
+  // Some text here
+  var intDividend = double.parse(dividend);
+  var intDivisor = double.parse(divisor);
+
+  return intDividend / intDivisor;
 }
+
+Color dynamicColor(String dividend, String divisor) {
```
Comment (Rey-Jey) [inline lib/BudgetCategories/budgetCategoryPrice.dart:17]: You could also set up the division outside of the function, that way it won't have to make a calculation every time it accesses the if statement.

Check out the way I did it in overallBudget.dart

File: lib/main.dart [hunk 6]
```diff
@@ -119,27 +119,14 @@ class HomePage extends StatelessWidget {
...
-                    )
-        ])),
+            child: ListView(
+                padding: const EdgeInsets.all(5),
+                children: const <Widget>[
+              SizedBox(height: 118, child: Center(child: HorProgressBar())),
+              SizedBox(height: 118, child: Center(child: HorProgressBar())),
+              SizedBox(height: 118, child: Center(child: HorProgressBar())),
+              SizedBox(height: 30, child: Center(child: addButton()))
+            ])),
```
Comment (Rey-Jey) [inline lib/main.dart:129]: Yeah, this organization is much better

**Review by Rey-Jey — APPROVED** [review]
The code works, there are just a couple things that could be tweaked

File: lib/ExpensesPage/expensesContainer.dart (hunk 1 above)
Comment (amuwhao46) [reply lib/ExpensesPage/expensesContainer.dart:3]: I see, I'll look through your changes and update mine accordingly

File: lib/BudgetCategories/budgetCategoryPrice.dart (hunk 4 above)
Comment (amuwhao46) [reply lib/BudgetCategories/budgetCategoryPrice.dart:5]: Jay is working on the forms for that, I want to get started on the variables after Jay does his (hopefully before the end of sprint 2) so we can actually test and see if it works. But I'm slowly making the skeleton for the methods and getters & setters

File: lib/BudgetCategories/budgetCategoryPrice.dart (hunk 5 above)
Comment (amuwhao46) [reply lib/BudgetCategories/budgetCategoryPrice.dart:17]: Sure thing, I'll make the update. The way I did it was an explicit change from string to double, thats why I left it like that

**Pull request merged**